pkg/profiler/cpu/cpu.go

```diff
@@ -158,6 +159,8 @@ func NewCPUProfiler(
 
 		memlockRlimit: memlockRlimit,
 
+		processErrorTracker: map[error]int{},
```

Comment:
I'm a bit worried about this being unbounded, let's use a capped cache implementation instead.